src/plugins/data/common/index_patterns/types.ts

```diff
@@ -34,3 +34,11 @@ export interface IIndexPattern {
     }
   >;
 }
+
+export interface IndexPatternAttributes {
```

Comment:
We can wait to get feedback from @kibana-app-arch on the following two things:
 - I think the data plugin follows the same convention as core to prepend all exported types with `I` so this should probably be`IIndexPatternAttributes`.
 - We probably want to discourage using these saved objects directly, so maybe this should be marked as deprecated with a note to rather use the index pattern service with the index pattern types.

Reply:
> We probably want to discourage using these saved objects directly, so maybe this should be marked as deprecated with a note to rather use the index pattern service with the index pattern types.

done.

> I think the data plugin follows the same convention as core to prepend all exported types with I so this should probably beIIndexPatternAttributes.

not sure. the same plugin https://github.com/elastic/kibana/blob/f18e7439d98a87f44487b46d13c426b5b1322685/src/plugins/data/common/query/types.ts#L20